tools/ds-identify

```diff
@@ -1486,16 +1526,16 @@ dscheck_VMware() {
         return "${DS_NOT_FOUND}"
     fi
 
-    # Do not proceed if the vmware-rpctool command is not present.
-    if ! vmware_has_rpctool; then
+    # Do not proceed if neither the vmware-rpctool or vmtoolsd command exists.
+    if ! vmware_has_rpctool ] && ! vmware_has_vmtoolsd; then
```

Comment:
There's still a close bracket left here (and I'm surprised it doesn't cause any failures)

Reply:
```shell
[0]akutz@akutzKFPN7:cloud-init $ ! false ] && ! true
[1]akutz@akutzKFPN7:cloud-init $ 
```

I am shocked as well (the number at the front of my prompt is the exit code from the previous command).

Reply:
Looks like y'all need to implement shellcheck.net for linting. I use it in all my projects for this very reason. Also note it's a warning and not an error. That alone shocks me...

<img width="792" alt="image" src="https://github.com/canonical/cloud-init/assets/101085/b87cdf1c-c6f6-4d92-8ea8-187ae775eab9">

Reply:
Hmmm...I thought we did. I have it in my editor and you'll notice some `# shellcheck disable` annotations thrown around ds-identify. I guess it never made it into CI though.

Reply:
https://github.com/canonical/cloud-init/issues/4453